numpy/core/src/multiarray/compiled_base.c

```diff
@@ -1402,7 +1402,11 @@ arr_add_docstring(PyObject *NPY_UNUSED(dummy), PyObject *args)
 {
     PyObject *obj;
     PyObject *str;
+    #if (PY_VERSION_HEX >= 0x030700A2)
+    const char *docstr;
+    #else
     char *docstr;
+    #endif
```

Comment:
What happens if you just always declare this `const`?

Reply:
Fails. I tried that earlier, and `const char *` warns on assignment to the docstring. The type of the latter seems to have also been changed in Python 3.7 with the addition of `const`.

Reply:
Could also fix this with a cast at the assignment location, but I guess fine to leave it.